### PR TITLE
ShutdownWorker frontend implementation

### PR DESCRIPTION
**Pull request by lina-temporal**
## What changed?
- ShutdownWorker is implemented in FrontendService by forwarding to `ForceUnloadTaskQueue` in Matching.

## Why?
- Will be used by workers to indicate their sticky queue is no longer being polled

## How did you test it?
- new test

## Potential risks
- new API that exposes `ForceUnloadTaskQueue`. If callers were to abuse the operation, they could force history being reloaded as the sticky queue is cleared.

## Documentation
- New docs on the API

## Is hotfix candidate?
- No

### CODE REVIEW

File: service/frontend/workflow_handler.go [hunk 1]
```diff
@@ -2608,6 +2608,25 @@ func (wh *WorkflowHandler) ResetStickyTaskQueue(ctx context.Context, request *wo
...
+	if request == nil {
+		return nil, errRequestNotSet
+	}
+
+	namespaceId, err := wh.namespaceRegistry.GetNamespaceID(namespace.Name(request.GetNamespace()))
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO: update poller info to indicate poller was shut down (pass identity/reason along)
```
Comment (carlydf) [inline service/frontend/workflow_handler.go:2620]: did you still want to do this TODO?

File: service/frontend/workflow_handler_test.go [hunk 2]
```diff
@@ -2943,3 +2944,30 @@ func (s *workflowHandlerSuite) TestExecuteMultiOperation() {
...
+	stickyTaskQueue := "sticky-task-queue"
+
+	expectedMatchingRequest := &matchingservice.ForceUnloadTaskQueueRequest{
+		NamespaceId:   s.testNamespaceID.String(),
+		TaskQueue:     stickyTaskQueue,
+		TaskQueueType: enumspb.TASK_QUEUE_TYPE_WORKFLOW,
+	}
+
+	s.mockNamespaceCache.EXPECT().GetNamespaceID(gomock.Eq(s.testNamespace)).Return(s.testNamespaceID, nil).AnyTimes()
+	s.mockMatchingClient.EXPECT().ForceUnloadTaskQueue(gomock.Any(), gomock.Eq(expectedMatchingRequest)).Return(&matchingservice.ForceUnloadTaskQueueResponse{}, nil)
```
Comment (carlydf) [inline service/frontend/workflow_handler_test.go:2962]: I'm assuming there are more comprehensive tests for `ForceUnloadTaskQueue` in the PR that introduced it?

**Review by carlydf — COMMENTED** [review]
LGTM, holding off on approval until I understand what the plan is for the TODO.

File: service/frontend/workflow_handler.go (hunk 1 above)
Comment (lina-temporal) [reply service/frontend/workflow_handler.go:2620]: Yeah, for now we're just using `ForceUnloadTaskQueue` as-is, so it doesn't note these.

File: service/frontend/workflow_handler_test.go (hunk 2 above)
Comment (lina-temporal) [reply service/frontend/workflow_handler_test.go:2962]: Yeah, I can backfill the tests on that.

Comment (lina-temporal) [reply service/frontend/workflow_handler_test.go:2962]: added `TestForceUnloadTaskQueue`

**Review by bergundy — APPROVED** [review]

**Review by Shivs11 — APPROVED** [review]
thank you for adding the test!

**Pull request merged**